src/main/java/ai/knowly/langtorch/preprocessing/splitter/text/TextSplitter.java

```diff
@@ -0,0 +1,134 @@
+                int numberOfIntermediateNewLines = 0;
+                if (prevChunk != null) {
+                    int indexChunk = StringUtils.indexOf(text, chunk);
+                    int indexEndPrevChunk = StringUtils.indexOf(text, prevChunk) + prevChunk.length();
+                    String removedNewlinesFromSplittingText = StringUtils.substring(text, indexChunk, indexEndPrevChunk);
+                    numberOfIntermediateNewLines = StringUtils.countMatches(removedNewlinesFromSplittingText, "\n");
+                }
+                lineCounterIndex += numberOfIntermediateNewLines;
+                int newLinesCount = StringUtils.countMatches(chunk, "\n");
+
+                Map<String, String> loc;
+                if (i < metadatas.getValues().size() && metadatas.getValues().get(i) != null) {
+                    if (!metadatas.getValues().get(i).isEmpty() && metadatas.getValues().get(i).get("loc") != null) {
+                        loc = new HashMap<>(metadatas.getValues().get(i));
+                    } else {
+                        loc = new HashMap<>();
+                    }
+                } else {
+                    loc = new HashMap<>();
+                }
+
+                loc.put("from", String.valueOf(lineCounterIndex));
+                loc.put("to", String.valueOf(lineCounterIndex + newLinesCount));
+
+                Map<String, String> metadataWithLinesNumber = new HashMap<>();
+                if (i < metadatas.getValues().size() && metadatas.getValues().get(i) != null) {
+                    metadataWithLinesNumber.putAll(metadatas.getValues().get(i));
+                }
+                metadataWithLinesNumber.putAll(loc);
+
+                documents.add(new DomainDocument(chunk, metadataWithLinesNumber));
+                lineCounterIndex += newLinesCount;
+                prevChunk = chunk;
+            }
+        }
+        return documents;
+    }
+
+    public List<DomainDocument> splitDocuments(List<DomainDocument> documents) {
+
+        List<DomainDocument> selectedDocs = documents.stream().filter(doc -> doc.getPageContent() != null).collect(Collectors.toList());
+
+        List<String> texts = selectedDocs.stream().map(DomainDocument::getPageContent).collect(Collectors.toList());
+        List<Map<String, String>> metaDatas = selectedDocs.stream().map(DomainDocument::getMetadata).collect(Collectors.toList());
+
+        return this.createDocuments(texts, Optional.of(new Metadatas(metaDatas)));
+    }
+
+    @Nullable
+    private String joinDocs(List<String> docs, String separator) {
+        String text = String.join(separator, docs);
+        return text.equals("") ? null : text;
+    }
+
+    public List<String> mergeSplits(List<String> splits, String separator) {
+        List<String> docs = new ArrayList<>();
+        List<String> currentDoc = new ArrayList<>();
+        int total = 0;
+
+        for (String d : splits) {
+            int _len = d.length();
```

Comment:
len = d.length();

we dont use _ to indicate it's a private variable like in python,